index.md

```diff
@@ -48,6 +48,361 @@ def is_even(num):
+type. Valid types are ints, slices, and objects with an __index__ method. For
+ example, inside a list, you should always remember to use the "index" to
+ indicate which element you want.  You should always keep in mind that in
+ csc108/148, a slice index could only be integer.
+
+```python
+a = ['p', 'y', 'T', 'A']
+print(a['p'])   # Error on this line
+```
+
+Corrected version
+
+```python
+a = ['p', 'y', 'T', 'A']
+print(a[0])      # Error on this line
+```
+
+### E1127: invalid-slice-index {#E1127}
+
+For your work in python, we expect you to use a slice index which is an integer,
+None, or an object with an __index__ method. For example, inside a list, whenever
+you want to get slices of this list, never use the elements but the "index". You
+should always keep in mind that in csc108/148, a slice index could only be
+integer.
+
+```python
+a = ['p', 'y', 'T', 'A']
+print(a['p' : 'A'])      # Error on this line
+```
+
+Corrected version
+
+```python
+a = ['p', 'y', 'T', 'A']
+print(a[0: 3])      # Error on this line
+```
+
+### E1128: assignment-from-none {#E1128}
+
+For your work in python, we do not expect to see when an assignment is done on a
+function call but the inferred function returns nothing but None. This error is
+similar to E1111. In the following example, f() return None. As a result, x has
+no value if we assign x f(). Remember! Whenever you define a function, return
+some objects otherwise it could not be assigned.
+
+```python
+def f():
+    return None
+
+
+def g():
+    x = f()  # Error on this line
+    print(x)
+```
+
+### E1130: invalid-unary-operand-type {#E1130}
+
+For your work in python, we expect you to use the unary operand on the objects
+which support this type of operation. For example, we should never add string
+with integer. You can convert the integer to str by using str(integer).
+Be careful when you try to add objects together.
+
+```python
+def info(age):
+    """
+    Return the info about my age.
+
+    @type age: int
+    @rtype: str
+    """
+    return "My age is" + age    # Error on this line
+                                # corrected version: "My age is" + str(age)
+info(2)
+```
+
+### E1131: unsupported-binary-operation {#E1131}
+
+For your work in python, we expect you to use two operands which support
+binary arithmetic operation. For example, a dictionary could not be added to
+a list. Make sure that you are adding objects which support binary arithmetic
+operation. This error may be a little similar to E1130.
+
+```python
+def my_fuc(t, n):
+    """Add <n> to <t>."""
+    if type(t) is tuple:
+        return t + (n, )
+    else:
+        return t + [n]  # Error on this line # thinks it is tuple
+
+my_fuc(1, 2)
+```
+
+### E1135: unsupported-membership-test {#E1135}
+
+For your work in python, we expect you to use an instance in membership test
+expression which implement membership protocol(__contains__/__iter__/__getitem__).
```

Comment:
Again, beginners don't know what "membership protocol" means, nor do they understand the special methods.

Also you should really begin by saying what the "membership test" actually is.